src/main/java/com/neroreflex/pizza/Pizza.java

```
@@ -91,79 +87,91 @@ public final String getBotID() {
     
     @Override
     protected void onMessage(String channel, String sender, String login, String hostname, String message) {
-        // No plugins che si chiamano da soli (o chiamano altri plugins)
-        if (sender.compareTo(this.getNick()) == 0) {
-            this.enqueueMessage(new Message(channel, "***CENSORED*** spam avoid"));
-            return;
-        }
+        // Ottengo il numero di caratteri nel nick
+        int nickLength = this.getNick().length();
         
-        // Analizza il messaggio per identificare richieste fatte al bot
-        Pattern invokeRegex = Pattern.compile("(" + this.getNick() + ")([\\s]+)([\\w]+)([\\s]*)([\\s\\S\\/\\:\\\\]*)");
-        Matcher invokeMatcher = invokeRegex.matcher(message);
-        if (invokeMatcher.matches()) {
+        // Cerco di capire se mi e' stato dato un comando
+        if ((message.charAt(0) == '!') && (message.length() > 1)) {
```

Può essere che arrivi un messaggio di lunghezza nulla?

---

https://tools.ietf.org/html/rfc2812.html Sezione 2.3.1: "Empty messages are silently ignored"